python/tvm/relay/pass_manager.py

```diff
@@ -0,0 +1,248 @@
+
+    pass_func : Callable[PassContext: tvm.relay.Module -> tvm.relay.Module]
+        The curried callback that sketches a certain optimization.
+    """
+
+    def __init__(self, name, opt_level, pass_func):
+        self.__init_handle_by_constructor__(_ir_pass.CreateModulePass, name,
+                                            opt_level, pass_func)
+
+    def __call__(self, mod):
+        """Execute a module pass.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module
+            The module that the module pass is executed on.
+
+        Returns
+        -------
+        ret : tvm.relay.Module
+            The updated module.
+        """
+        return _ir_pass.RunModulePass(self, mod)
+
+
+@register_relay_node
+class FunctionPass(Pass):
+    """A pass that works on each tvm.relay.Function in a module.
+
+    Parameters
+    ----------
+    name : str
+        The pass name.
+
+    opt_level : int
+        The optimization level of this pass.
+
+    pass_func : Callable[PassContext: tvm.relay.Function -> tvm.relay.Function]
+        The curried callback that sketches a certain optimization.
+    """
+
+    def __init__(self, name, opt_level, pass_func):
+        self.__init_handle_by_constructor__(_ir_pass.CreateFunctionPass, name,
+                                            opt_level, pass_func)
+
+    def __call__(self, mod):
+        """Execute a function pass.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module
+            The module that the function pass is executed on.
+
+        Returns
+        -------
+        ret : tvm.relay.Module
+            The updated module.
+        """
+        return _ir_pass.RunFunctionPass(self, mod)
+
+
+@register_relay_node
+class SequentialPass(Pass):
+    """A pass that works on each tvm.relay.Function in a module.
+
+    Parameters
+    ----------
+    name : str
+        The pass name.
+
+    opt_level : int
+        The optimization level of this pass.
+
+    passes : List[Pass]
+        The pass candidates to be executed.
+
+    disabled : Optional[List[str]]
+        The list of passes that are disabled.
+    """
+
+    def __init__(self, name, opt_level, passes, disabled=None):
+        disabled = disabled if disabled else []
+        if not isinstance(disabled, (list, tuple)):
+            raise TypeError("disabled must be a list or tuple of pass names")
+        self.__init_handle_by_constructor__(_ir_pass.CreateSequentialPass,
+                                            name, opt_level, passes, disabled)
+
+    def __call__(self, mod):
+        """Execute a sequence of passes.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module
+            The module that the function pass is executed on.
+
+        Returns
+        -------
+        ret : tvm.relay.Module
+            The updated module.
+        """
+        return _ir_pass.RunSequentialPass(self, mod)
+
+
+def create_pass(pass_name, opt_level,
```

Comment:
let us try to make things consistent with C++.

Perhaps do ir_pass.create_sequential_pass etc.